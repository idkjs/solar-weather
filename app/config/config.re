let units = ["c", "f"];

let timeTypes = ["24", "12"];

module AppColors = {
  let red = "#E77F6D";
  let opaqueBlack = "rgba(0, 0, 0, 0.2)";
  let opaqueBlack70 = "rgba(0, 0, 0, 0.7)";
  let lightGrey = "#EFEFEF";
  let white = "#FFF";
  let whiteGrey = "#F9F9F9";
  let medGrey = "#F0F0F0";
  let black = "#000";
  let darkGrey = "#343434";
  let grey = "#999";
  let medGreyAlt = "#C0C0C0";
};

let drawerTweenHandler = (ratio: float) => {
  "drawerOverlay": {opacity: (1. -. ratio) /. 2.},
  "mainOverlay": {opacity: ratio *. 0.1}
};

let drawerStyles = {
  "drawerOverlay": {backgroundColor: "#000", opacity: 1.},
  "mainOverlay": {backgroundColor: "#000", opacity: 0.}
};
