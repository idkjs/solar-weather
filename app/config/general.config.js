const units = ['c', 'f'];
const timeTypes = ['24', '12'];

const appColors = {
  red: '#E77F6D',
  opaqueBlack: 'rgba(0, 0, 0, 0.2)',
  opaqueBlack70: 'rgba(0, 0, 0, 0.7)',
  lightGrey: '#EFEFEF',
  white: '#FFF',
  whiteGrey: '#F9F9F9',
  medGrey: '#F0F0F0',
  black: '#000000',
  darkGrey: '#343434',
  grey: '#999',
  medGreyAlt: '#C0C0C0',
};

module.exports = {
  units,
  timeTypes,
  appColors,
};
