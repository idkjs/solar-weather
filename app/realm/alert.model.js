class Alert {}
Alert.schema = {
  name: 'Alert',
  properties: {
    title: 'string',
    time: 'int',
    severity: 'string',
    expires: 'int',
    description: 'string',
    uri: 'string',
  },
};

export default Alert;
