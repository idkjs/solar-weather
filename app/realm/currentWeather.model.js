class Currently {}
Currently.schema = {
	name: 'Currently',
	properties: {
		summary: 'string',
		icon: 'string',
		precipIntensity: 'float',
		precipProbability: 'float',
		precipType: { type: 'string', optional: true },
		temperature: 'float',
		humidity: 'float',
		apparentTemperature: 'float',
	},
};

export default Currently;
