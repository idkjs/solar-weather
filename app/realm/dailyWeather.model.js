class DailyData {}
DailyData.schema = {
  name: 'DailyData',
  properties: {
    time: 'int',
    summary: 'string',
    icon: 'string',
    sunriseTime: 'int',
    sunsetTime: 'int',
    precipIntensity: 'float',
    precipProbability: 'float',
    precipType: { type: 'string', optional: true },
    temperatureMin: 'float',
    temperatureMax: 'float',
    humidity: 'float',
  },
};

class Daily {}
Daily.schema = {
  name: 'Daily',
  properties: {
    summary: 'string',
    icon: 'string',
    data: { type: 'list', objectType: 'DailyData' },
  },
};

module.exports = {
  Daily,
  DailyData,
};
