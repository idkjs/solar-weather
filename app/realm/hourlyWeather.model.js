class HourlyData {}
HourlyData.schema = {
	name: 'HourlyData',
	properties: {
		time: 'int',
		summary: 'string',
		icon: 'string',
		precipIntensity: 'float',
		precipProbability: 'float',
		precipType: { type: 'string', optional: true },
		temperature: 'float',
		humidity: 'float',
	},
};

class Hourly {}
Hourly.schema = {
	name: 'Hourly',
	properties: {
		summary: 'string',
		icon: 'string',
		data: { type: 'list', objectType: 'HourlyData' },
	},
};

module.exports = {
	Hourly,
	HourlyData,
};
