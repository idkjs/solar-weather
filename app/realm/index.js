import Realm from 'realm';

import Locations from './locations.model';
import Options from './options.model';
import { Daily, DailyData } from './dailyWeather.model';
import Alert from './alert.model';
import { Hourly, HourlyData } from './hourlyWeather.model';
import Currently from './currentWeather.model';

const realm = new Realm({
	schema: [
		Locations,
		Alert,
		Options,
		Daily,
		DailyData,
		Hourly,
		HourlyData,
		Currently,
	],
	schemaVersion: 78,
	deleteRealmIfMigrationNeeded: true,
});

export const deleteRealm = async () => {
	try {
		await realm.write(async () => {
			realm.deleteAll();
		});
	} catch (err) {
		if (err) Promise.reject(err);
	}
};

export default realm;
