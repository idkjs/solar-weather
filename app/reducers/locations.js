import moment from 'moment';
import * as types from '../actions/types.action';

export const initialState = (date) => {
  return {
    locations: [{
      id: 1,
      lat: 0.0,
      lng: 0.0,
      currently: {
        icon: '',
        precipProbability: 0.00,
        precipType: '',
        temperature: 0.00,
        apparentTemperature: 0.00,
        humidity: 0,
        summary: '',
      },
      daily: {
        data: [],
      },
      hourly: {
        data: [],
      },
      alerts: [],
    }],
    timezone: 'America/New_York',
    locationError: null,
    loading: false,
    latestCollectiveUpdate: date || moment().subtract(1, 'day'),
  };
};

export default function locations(state = initialState(), action = {}) {
  switch (action.type) {
    case types.LOCATION_LOADING_OFF:
      return {
        ...state,
        loading: false,
      };
    case types.FETCH_LOCATIONS:
    case types.LOCATION_LOADING:
      return {
        ...state,
        loading: true,
      };
    case types.GET_LOCATIONS:
      return {
        ...state,
        locations: action.locations,
        locationError: null,
        loading: false,
      };
    case types.SET_LOCATION: // eslint-disable-line
      const currentLocation = Object.assign(state.locations[action.index], action.location);
      const locs = state.locations;
      locs[action.index] = currentLocation;
      return {
        ...state,
        locations: locs,
        locationError: null,
        loading: false,
      };
    case types.ADD_LOCATION:
      return {
        ...state,
        locations: [...state.locations.filter(x => x.id !== 1), action.location],
        locationError: null,
        loading: false,
      };
    case types.FETCH_LOCATIONS_SUCCESS:
      return {
        ...state,
        locations: action.locations,
        locationError: null,
        latestCollectiveUpdate: moment(),
        loading: false,
      };
    case types.DELETE_LOCATION: // eslint-disable-line
      return {
        ...state,
        locations: [...state.locations.filter(x => x.id !== action.id)],
        locationError: null,
        loading: false,
      };
    case types.FETCH_LOCATIONS_FAILURE:
    case types.ADD_INDEX_LOCATION_ERROR:
    case types.ADD_LOCATION_ERROR:
    case types.SET_ACTIVE_LOCATION_ERROR:
    case types.UPDATE_ERROR:
      return {
        ...state,
        locationError: action.err,
        loading: false,
      };
    default:
      return state;
  }
}
