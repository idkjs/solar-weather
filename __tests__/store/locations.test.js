import mockStore from 'redux-mock-store';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { deleteLocationFromStore, addNewLocation } from '../../app/actions/locations.action';
import { loc } from '../config';
import { deleteRealm } from '../../app/realm';

const mockApi = new MockAdapter(axios);
// Mock request to/from DarkSky
mockApi.onGet(/\/forecast\//).reply(200, loc);

describe('locations store', () => {
  beforeAll(async () => {
    jest.genMockFromModule('realm');
    // Teardown Realm prior to running tests
    await deleteRealm();
  });

  afterAll(async () => {
    // Clean up after adding to Realm
    await deleteRealm();
  });

  beforeEach(() => {
    store.clearActions();
  });

  const store = mockStore();
  it('should handle deleteLocationFromStore', async () => {
    await store.dispatch(deleteLocationFromStore(0));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should handle faulty location by returning error', async () => {
    await store.dispatch(addNewLocation({ lat: 0.0, lng: 0.0 }, 1));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should handle correct location by returning success', async () => {
    await store.dispatch(addNewLocation({ lat: 35.689487, lng: 139.691706, name: 'Tokyo' }, 0));
    expect(store.getActions()).toMatchSnapshot();
  });
});
