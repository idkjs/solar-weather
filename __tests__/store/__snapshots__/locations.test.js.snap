// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locations store should handle correct location by returning success 1`] = `
Array [
  Object {
    "type": "LOCATION_LOADING",
  },
  Object {
    "location": Object {
      "alerts": Array [],
      "currently": Object {
        "apparentTemperature": 0,
        "humidity": 0,
        "icon": "sunny",
        "precipIntensity": 0,
        "precipProbability": 0,
        "precipType": "rain",
        "summary": "Test summary",
        "temperature": 0,
      },
      "daily": Object {
        "data": Array [],
        "icon": "sunny",
        "summary": "Test summary",
      },
      "hourly": Object {
        "data": Array [],
        "icon": "sunny",
        "summary": "Test summary",
      },
      "id": 1515148913531,
      "last_updated": 2018-01-05T10:41:53.531Z,
      "lat": 35.689487,
      "lng": 139.691706,
      "name": "Tokyo",
      "timezone": "America/New_York",
    },
    "type": "ADD_LOCATION",
  },
  Object {
    "index": 0,
    "type": "SET_ACTIVE_LOCATION",
  },
]
`;

exports[`locations store should handle deleteLocationFromStore 1`] = `
Array [
  Object {
    "id": 0,
    "type": "DELETE_LOCATION",
  },
  Object {
    "index": 0,
    "type": "SET_ACTIVE_LOCATION",
  },
]
`;

exports[`locations store should handle faulty location by returning error 1`] = `
Array [
  Object {
    "type": "LOCATION_LOADING",
  },
  Object {
    "err": "Error adding location",
    "type": "UPDATE_ERROR",
  },
]
`;
